### ISSUE TITLE
Is getting the mAP = 0.0000602 the correct value for the Accuracy Performance?

### ISSUE THREAD

**Issue by jaskiratsingh2000** [ultralytics/yolov5#3839]
Hi @glenn-jocher I tried running the test.py to get the metric and mAP value. The **mAP** value I got is **0.0000602** can you please let me know that what is the best value and wh I am getting this value on my machine?

**Comment by glenn-jocher** [comment]
@jaskiratsingh2000 valid mAP values are between 0 and 1.

**Comment by jaskiratsingh2000** [comment]
@glenn-jocher that I know actually but what I mean to ask is that why I am getting this much low valu because the value you mentionedin the repo is around 0.682 and I am getting 0.0000602 ? What can be the reason for getting this much low value?
@glenn-jocher let me know please

**Comment by glenn-jocher** [comment]
@jaskiratsingh2000 

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

* ✅ **Minimal** – Use as little code as possible that still produces the same problem
* ✅ **Complete** – Provide **all** parts someone else needs to reproduce your problem in the question itself
* ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

In addition to the above requirements, for [Ultralytics](https://ultralytics.com/) to provide assistance your code should be:

* ✅ **Current** – Verify that your code is up-to-date with current GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been resolved by previous commits.
* ✅ **Unmodified** – Your problem must be reproducible without any modifications to the codebase in this repository. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all of the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) and providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by jaskiratsingh2000** [comment]
@glenn-jocher Thanks for sharing this. I am actually asking the reason of getting such low value. You haven't told me yet that what can be the reason for getting such mAP value.
Once you rectify, I am happy to open up the bug report if that so.
Thanks

**Comment by tetelevm** [comment]
@jaskiratsingh2000   If I understood correctly, the answer above was that without the data of your atmosphere and gravity it is impossible to say why your horse in your conditions is not spherical, but in the form of a pyramid. You need to give data that can answer your question (training parameters, images, number of classes...).

**Comment by jaskiratsingh2000** [comment]
@tetelevm @glenn-jocher I didn't use the custom data instead I use everything pre-trained that is COCO data set and it is this one https://github.com/ultralytics/yolov5/blob/master/data/coco.yaml

The command i ran was :

`python3 test.py --data coco.yaml --weights yolov5s.pt`

@tetelevm @glenn-jocher let me know please if you can answer it now

**Comment by jaskiratsingh2000** [comment]
@tetelevm @glenn-jocher please let me know on above. I used COCO dataset only nothing else.

**Comment by tetelevm** [comment]
@jaskiratsingh2000 
(if anything, I have nothing to do with the development of YOLO, I am a user just like you)
What were the parameters for training the network?

**Comment by tetelevm** [comment]
As I understand it, the pre-trained network has no weights on one or more last layers, and the main point of training on the pre-trained network is just to train the last layers. If you used just a pre-trained model without further training, it should be `result = random()` for each image.

 @glenn-jocher Please comment on this thought, I am interested in the actual state of affairs.

**Comment by jaskiratsingh2000** [comment]
@tetelevm I didn't train anything. As I mentioned I just used the above command to figure out the metric.

How can I take it to the same mAP value @tetelevm? 

I actually tried previous test with training of 400 images but it just gave me everything 0. Even the mAP value 0. Why so?

**Comment by jaskiratsingh2000** [comment]
@glenn-jocher @tetelevm can you please let me know?

**Comment by tetelevm** [comment]
I'm not sure why that is, my guess is that using a pre-trained model without training is wrong, and here's why.

![img](https://user-images.githubusercontent.com/46541662/123977343-70817000-d9ae-11eb-923c-e270c54e85bd.png)

My guess is probably correct, but someone more knowledgeable would be able to tell for sure.

**Comment by jaskiratsingh2000** [comment]
@tetelevm as I mentioned alongside pre-trained dataset I used pre-trained weights as 'yolovts.pt'

@tetelevm If you want me to train for scratch can you please let me know that what parameters I should use dor training the network like weights, epochs, etc etc to get the correct accuracy performance?

**Comment by tetelevm** [comment]
They made you a very high quality dataset, created a very accurate model, put optimal hyperparameters in it, that is, they did 98% of the work for you. All you have to do is read the first two `README` tutorials and use them.

And also, a tip - before you use YOLO, make a neural network in pure PyTorch, which, for example, will distinguish a pen from a pencil.

**Comment by jaskiratsingh2000** [comment]
@glenn-jocher @tetelevm  I custom trained the dataset on 1000 images and I just got the 0 mAP value. Not sure why? Can you please let me know, please?

**Comment by tetelevm** [comment]
@jaskiratsingh2000 Please give me an example of the code with which you ran the training.

**Comment by jaskiratsingh2000** [comment]
@tetelevm I used this dataset in yolov5 pytorch format - https://public.roboflow.com/object-detection/mask-wearing/4/download

for training I used the following command `python3 train.py --data ./data.yaml --img 416 --batch 80 --epochs 100 --cfg yolov5s.yaml  --weights yolov5s.pt`

@tetelevm Let me know if I need to improve in some way.

**Comment by tetelevm** [comment]
@jaskiratsingh2000 If you are sure that the data are correct (you can check this with any data markup program, for example I use [labelImg](https://github.com/tzutalin/labelImg)), and also that you have correct class/path/... parameters, then already at epoch 10 you will see that the model is not learning (with my not very good data the pre-trained `m`-version learns up to 0.97 in 20 epochs).

**Comment by tetelevm** [comment]
> And also, a tip - before you use YOLO, make a neural network in pure PyTorch, which, for example, will distinguish a pen from a pencil.

That's really good advice. People who try to write complicated things in Django but don't even know OOP in Python won't do anything serious. Same here, if you understand how to build and train models in Pytorch, you'll have a much easier time figuring out YOLO as well.

**Comment by jaskiratsingh2000** [comment]
@televlem that's okay but I don't want to go with the pytorch thing from scract due to less time.

So can You let me know that on what datset(already labelled one) I can use what parameters?

Also can I train already existing COCO dataset with 5 epochs?

**Comment by tetelevm** [comment]
@jaskiratsingh2000  I'm not sure, but try 20-30 epochs (in neighboring questions they say that after 30 epochs they go overtraining).

**Comment by jaskiratsingh2000** [comment]
@tetelevm Why do you think that I should retrain? Does it affect in any sense? I just need to evaluate the performance metric

**Comment by tetelevm** [comment]
What do you want to retrain? You don't need to `retrain`, you need to `train` a pre-trained model with the command
```
python train.py --img 416 --batch 100 --epochs 20 --data coco128.yaml --weights yolov5m.pt --cfg yolov5m.yaml --name MY_YOLO
```
And then take the file `runs/train/MY_YOLO/weigts/best.pt` and test on it.

**Comment by jaskiratsingh2000** [comment]
okay let me try that

**Comment by jaskiratsingh2000** [comment]
@tetelevm Even after doing the above training of the pre-trained model I got the almost same accuracy performance. I am not sure why it is happening. FYI, I am testing this out on Raspberry Pi

**Comment by tetelevm** [comment]
`testing this out on Raspberry Pi` There should be no difference, a computer is a computer. 

If the accuracy doesn't change, I don't know how to help you anymore.

The success of learning depends on two things - the data and the algorithm. If you do everything exactly according to the tutorial, there should be nothing wrong with the algorithm, and I would assume that the problem is with the data. But if you're also using the standard COCO dataset from the same tutorial, then I don't know what might be wrong, you'll have to figure it out for yourself.

**Comment by jaskiratsingh2000** [comment]
@tetelevm Okay so I worked with the way you mentioned above which is training on the pre-trained model. I used the following to train on my desktop `python train.py --img 416 --batch 300 --epochs 50 --data coco.yaml --weights yolov5s.pt --cfg yolov5s.yaml --name MY_YOLO` and tested the same on the Raspberry Pi but again that gave me 0.000067 mAP value and I am not really sure why it has happened. Can you please try helping me if possible @glenn-jocher @tetelevm ?

**Comment by jaskiratsingh2000** [comment]
@tetelevm any update on this yet? - https://docs.google.com/document/d/1ej8hIktTFmsqTH0GCYVw-qD8sF9C9fa_Pe61EenNmkU/edit?usp=sharing
